### PR TITLE
Fixes #7928: Merge common from initial promises and system technique

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7928

### CODE REVIEW

**Comment by peckpeck** [comment]
PR updated

File: initial-promises/node-server/common/1.0/rudder_lib.cf [hunk 1]
```diff
@@ -658,7 +655,41 @@ bundle edit_line insert_rudder_disclaimer
 {
   insert_lines:
 
-      "## File managed by Rudder - Edit with caution, some changes may disappear (especially passwords) ##"
+      "${rudder_parameters.rudder_file_edit_header}"
```
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/rudder_lib.cf]: This won't work unless you also include the rudder_parameters.cf file

File: initial-promises/node-server/common/1.0/site.cf [hunk 2]
```diff
@@ -44,78 +44,95 @@ bundle common g
...
-      "uncompressed_inventory_file_types"  slist => { ".*.ocs" };
-      "inventory_file_types"               slist => { "@{uncompressed_inventory_file_types}", ".*.gz" };
-      "signed_inventory_file_types"        slist => { "@{inventory_file_types}", ".*\.sign" };
-
-      "server_roles_path"                 string => "${rudder_base}/etc/server-roles.d/";
+      "uuid"                        string => readfile("${g.uuid_file}", 60);
+      "server_shares_folder"        string => "/var/rudder/share/${uuid}/promises/shares";
+      "rudder_var_reports"          string => "${rudder_var}/reports";
+      "davuser"                     string => "";
+      "davpw"                       string => "";
```
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: Looks like these values are missing from your variables.json?

File: initial-promises/node-server/common/1.0/site.cf [hunk 3]
```diff
@@ -44,78 +44,95 @@ bundle common g
...
+      "minicurl"                    string => "${rudder_bin}/rudder-perl ${sys.workdir}/inputs/common/utilities/minicurl";
+      "excludedreps"                slist  => { "\.X11", ".*kde.*", "\.svn", "perl" };
+      "ncf_local_excludedreps"      slist => { "50_techniques", @{g.excludedreps}};
+      "rudder_tools_origin"         string => "/var/rudder/tools";
+
+      # Path where all the configurations (git with techniques, groups, ncf, etc) are stored
+      "rudder_configuration_repository"
+                                    string => "";
```
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: Same here

File: initial-promises/node-server/common/1.0/site.cf [hunk 4]
```diff
@@ -44,78 +44,95 @@ bundle common g
...
       # We would like to use date's "--rfc-3339=second" option here, but it is not available on older OSes (RHEL 3/4, AIX 5...)
-      "execRun" string => execresult("/bin/date -u \"+%Y-%m-%d %T+00:00\"", "noshell");
+      "execRun"                    string => execresult("/bin/date -u \"+%Y-%m-%d %T+00:00\"", "noshell");
+
+      # Heartbeat interval, in minutes. Only used when in "changes-only" compliance mode
+      # In this mode, we should only send a "StartRun"/"EndRun" message pair if changes occur,
+      # or if this interval has elapsed since we last sent one
+      "heartbeat_interval" string => "";
+
+      "rudder_node_config_id" string => "";
```
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: And these 2 (heartbeat_interval and rudder_node_config_id=

File: initial-promises/node-server/common/1.0/site.cf (hunk 2 above)
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: Actually I see they are in the variables.json file, but for some reason they have not appeared here.

File: initial-promises/node-server/common/1.0/site.cf (hunk 3 above)
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: This one however is actually missing from variables.json

File: initial-promises/node-server/common/1.0/site.cf (hunk 4 above)
Comment (jooooooon) [inline initial-promises/node-server/common/1.0/site.cf]: These 2 are in the same case as davuser and davpw, they are in variables.json but for some reason have not appeared in this generated file.

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Comment by peckpeck** [comment]
PR updated

**Pull request merged**